state_synchronizer/src/peer_manager.rs

```diff
@@ -92,30 +99,57 @@ impl PeerManager {
 
     pub fn update_score(&mut self, peer_id: &PeerId, update_type: PeerScoreUpdateType) {
         if let Some(peer_info) = self.peers.get_mut(peer_id) {
+            let mut is_update = false;
```

Comment:
why not just do `let old_score =  peer_info.score`
and then later compare that they are not equal?